src/program/mixins.py

```diff
@@ -12,20 +12,17 @@
 
 
 class EnsureCFPOpenMixin:
-    def dispatch(self, request, *args, **kwargs):
+    def setup(self, request, *args, **kwargs):
         # do not permit this action if call for participation is not open
         if not self.camp.call_for_participation_open:
             messages.error(request, "The Call for Participation is not open.")
             return redirect(
```

Comment:
we cannot return a HttpResponse from inside setup(), only dispatch() can do that. Either raise an exception or keep this as dispatch()